### PR TITLE
core: message redaction for auditing

### PR DESCRIPTION
**Pull request by danielhochman**
### Description

Adds a "redacted" message annotation which audit respects on request/response objects in order to log a redacted placeholder.

This prevents any sensitive data from being logged in auditing.

Another possibility would be moving the annotation to individual fields, but it's significantly more complicated.

### Testing Performed
TODO

### TODOs
- [x] Unit testing

### CODE REVIEW

**Review by mcutalo88 — COMMENTED** [review]
Nice looks good, just needs some tests.

File: backend/gateway/meta/meta_test.go [hunk 1]
```diff
@@ -118,6 +120,23 @@ func TestResourceNames(t *testing.T) {
 	}
 }
 
+func TestIsRedacted(t *testing.T) {
+	assert.True(t, IsRedacted(&authnv1.CallbackResponse{}))
+	assert.False(t, IsRedacted(&healthcheckv1.HealthcheckRequest{}))
+}
+
+func TestAPIBodyRedaction(t *testing.T) {
```
Comment (dschaller) [inline backend/gateway/meta/meta_test.go:128]: Should we have a test for login response as well just to be sure the field isn't removed in the future?

**Review by dschaller — APPROVED** [review]
LGTM, pending question

File: backend/gateway/meta/meta_test.go (hunk 1 above)
Comment (danielhochman) [reply backend/gateway/meta/meta_test.go:128]: sure, i'll add tests to the auditing module itself to check this

**Review by mcutalo88 — APPROVED** [review]

**Pull request merged**